pkg/storage/unified/search/bleve_index_metrics.go

```diff
@@ -0,0 +1,141 @@
+	onceIndex.Do(func() {
+		IndexMetrics = &BleveIndexMetrics{
+			IndexDir: indexDir,
+			Backend:  bleveBackend,
+			IndexLatency: prometheus.NewHistogramVec(prometheus.HistogramOpts{
+				Namespace:                       "index_server",
+				Name:                            "index_latency_seconds",
+				Help:                            "Time (in seconds) until index is updated with new event",
+				Buckets:                         instrument.DefBuckets,
+				NativeHistogramBucketFactor:     1.1, // enable native histograms
+				NativeHistogramMaxBucketNumber:  160,
+				NativeHistogramMinResetDuration: time.Hour,
+			}, []string{"resource"}),
+			IndexSize: prometheus.NewGauge(prometheus.GaugeOpts{
+				Namespace: "index_server",
+				Name:      "index_size",
+				Help:      "Size of the index in bytes - only for file-based indices",
+			}),
+			IndexedDocs: prometheus.NewGauge(prometheus.GaugeOpts{
+				Namespace: "index_server",
+				Name:      "indexed_docs",
+				Help:      "Number of indexed documents by resource",
+			}),
+			IndexedKinds: prometheus.NewGaugeVec(prometheus.GaugeOpts{
+				Namespace: "index_server",
+				Name:      "indexed_kinds",
+				Help:      "Number of indexed documents by kind",
+			}, []string{"kind"}),
+			IndexCreationTime: prometheus.NewHistogramVec(prometheus.HistogramOpts{
+				Namespace:                       "index_server",
+				Name:                            "index_creation_time_seconds",
+				Help:                            "Time (in seconds) it takes until index is created",
+				Buckets:                         IndexCreationBuckets,
+				NativeHistogramBucketFactor:     1.1, // enable native histograms
+				NativeHistogramMaxBucketNumber:  160,
+				NativeHistogramMinResetDuration: time.Hour,
+			}, []string{}),
+			IndexTenants: prometheus.NewCounterVec(prometheus.CounterOpts{
+				Namespace: "index_server",
+				Name:      "index_tenants",
+				Help:      "Number of tenants in the index",
+			}, []string{"slug", "index_storage"}), // index_storage is either "file" or "memory"
+		}
+	})
+
+	return IndexMetrics
+}
+
+func (s *BleveIndexMetrics) Collect(ch chan<- prometheus.Metric) {
+	s.IndexLatency.Collect(ch)
+	s.IndexCreationTime.Collect(ch)
+	s.IndexedKinds.Collect(ch)
+	s.IndexTenants.Collect(ch)
+
+	// collect index size
+	totalSize, err := getTotalIndexSize(s.IndexDir)
+	if err == nil {
+		s.IndexSize.Set(float64(totalSize))
+		s.IndexSize.Collect(ch)
+	}
+
+	// collect index docs
+	s.IndexedDocs.Set(s.getTotalDocCount())
+	s.IndexedDocs.Collect(ch)
+}
+
+func (s *BleveIndexMetrics) Describe(ch chan<- *prometheus.Desc) {
+	s.IndexLatency.Describe(ch)
+	s.IndexSize.Describe(ch)
+	s.IndexedDocs.Describe(ch)
+	s.IndexedKinds.Describe(ch)
+	s.IndexCreationTime.Describe(ch)
+	s.IndexTenants.Describe(ch)
+}
+
+// getTotalDocCount returns the total number of documents in the index
+func (s *BleveIndexMetrics) getTotalDocCount() float64 {
+	var totalDocs float64
+	for _, v := range s.Backend.cache {
```

Comment:
@ryantxu this is why I made the BleveBackend public. This is the only place in the metrics where its coupled to Bleve, so I suppose we could add a totalDocs() function onto the StorageBackend interface, and use that here instead?

Reply:
something like this https://github.com/grafana/grafana/pull/97310/commits/7921bd1a50b478c940a727c68d89b755f575986c